tests/aws/services/stepfunctions/v2/callback/test_callback.py

```diff
@@ -773,3 +776,78 @@ def _get_message_body():
             definition,
             exec_input,
         )
+
+    @markers.aws.validated
+    def test_sync_with_task_token(
+        self,
+        aws_client,
+        sqs_create_queue,
+        sqs_send_task_success_state_machine,
+        create_iam_role_for_sfn,
+        create_state_machine,
+        sfn_snapshot,
+    ):
+        sfn_snapshot.add_transformer(
+            JsonpathTransformer(
+                jsonpath="$..output.StartDate",
+                replacement="start-date",
+                replace_reference=False,
+            )
+        )
+        sfn_snapshot.add_transformer(
+            JsonpathTransformer(
+                jsonpath="$..output.StopDate",
+                replacement="stop-date",
+                replace_reference=False,
+            )
+        )
+        sfn_snapshot.add_transformer(
+            JsonpathTransformer(
+                jsonpath="$..MessageId",
+                replacement="message_id",
+                replace_reference=True,
+            )
+        )
+        sfn_snapshot.add_transformer(
+            JsonpathTransformer(
+                jsonpath="$..TaskToken",
+                replacement="task_token",
+                replace_reference=True,
+            )
+        )
+        sfn_snapshot.add_transformer(sfn_snapshot.transform.sfn_sqs_integration())
+
+        queue_name = f"queue-{short_uid()}"
+        queue_url = sqs_create_queue(QueueName=queue_name)
+        sfn_snapshot.add_transformer(RegexTransformer(queue_url, "sqs_queue_url"))
+        sfn_snapshot.add_transformer(RegexTransformer(queue_name, "sqs_queue_name"))
+
+        sqs_send_task_success_state_machine(queue_url)
```

Comment:
I initially missed this important step and was wondering how/why the StepFunction gets terminated. Adding a comment would be helpful here because that fixture hides quite a complex process essential for understanding the test case. The fixture itself also doesn't have a human-readable summary.